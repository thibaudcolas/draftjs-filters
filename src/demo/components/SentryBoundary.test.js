import React from "react"
import { shallow } from "enzyme"
import SentryBoundary from "./SentryBoundary"

describe("SentryBoundary", () => {
  it("renders", () => {
    expect(shallow(<SentryBoundary>Test</SentryBoundary>)).toMatchSnapshot()
  })

  it("componentDidCatch", () => {
    const wrapper = shallow(<SentryBoundary>Test</SentryBoundary>)

    wrapper.instance().componentDidCatch(new Error("test"), {
      info: "test",
    })

    expect(wrapper.state("error")).not.toBe(null)
  })

  it("componentDidCatch Raven", () => {
    window.Raven = {
      captureException: jest.fn(),
    }
    const wrapper = shallow(<SentryBoundary>Test</SentryBoundary>)

    wrapper.instance().componentDidCatch(new Error("test"), {
      info: "test",
    })

    expect(window.Raven.captureException).toHaveBeenCalled()
  })

  it("#error", () => {
    window.Raven = false
    expect(
      shallow(<SentryBoundary>Test</SentryBoundary>).setState({
        error: new Error("test"),
      }),
    ).toMatchSnapshot()
  })

  it("#error reload", () => {
    window.Raven = false
    window.location.reload = jest.fn()

    shallow(<SentryBoundary>Test</SentryBoundary>)
      .setState({
        error: new Error("test"),
      })
      .find("button")
      .simulate("click")
    expect(window.location.reload).toHaveBeenCalled()
  })

  it("#error Raven", () => {
    window.Raven = true
    expect(
      shallow(<SentryBoundary>Test</SentryBoundary>).setState({
        error: new Error("test"),
      }),
    ).toMatchSnapshot()
  })

  it("#error Raven report", () => {
    window.Raven = {
      lastEventId: jest.fn(() => true),
      showReportDialog: jest.fn(),
    }
    window.location.reload = jest.fn()

    shallow(<SentryBoundary>Test</SentryBoundary>)
      .setState({
        error: new Error("test"),
      })
      .find("button")
      .at(0)
      .simulate("click")
    expect(window.Raven.lastEventId).toHaveBeenCalled()
  })
})
