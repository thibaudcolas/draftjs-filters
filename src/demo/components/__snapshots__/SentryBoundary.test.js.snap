// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentryBoundary #error 1`] = `
<div
  className="DraftEditor-root"
>
  <div
    className="DraftEditor-editorContainer"
  >
    <div
      className="public-DraftEditor-content"
    >
      <div
        className="u-text-center"
      >
        <p>
          Oops. The editor just crashed.
        </p>
        <p>
          Our team has been notified. You can provide us with more information if you want to.
        </p>
        <div>
          <a
            href="https://github.com/thibaudcolas/draftjs-filters/issues"
            rel="noopener noreferrer"
            style={
              Object {
                "textDecoration": "underline",
              }
            }
            target="_blank"
          >
            Open a GitHub issue
          </a>
          <span>
             
          </span>
          <button
            onClick={[Function]}
            type="button"
          >
            Reload the page
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SentryBoundary #error Raven 1`] = `
<div
  className="DraftEditor-root"
>
  <div
    className="DraftEditor-editorContainer"
  >
    <div
      className="public-DraftEditor-content"
    >
      <div
        className="u-text-center"
      >
        <p>
          Oops. The editor just crashed.
        </p>
        <p>
          Our team has been notified. You can provide us with more information if you want to.
        </p>
        <div>
          <button
            onClick={[Function]}
            type="button"
          >
            Submit a report
          </button>
          <span>
             
          </span>
          <button
            onClick={[Function]}
            type="button"
          >
            Reload the page
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SentryBoundary renders 1`] = `"Test"`;
